src/components/editor/shapeComponents/SetDefinitionBuilder.vue

```diff
@@ -153,20 +167,28 @@ if (props.shape.argument?.some(arg => arg.valueVariable) && valueVariableMap) {
   );
 }
 
-const key = props.shape.path.iri;
-const buttonOptions = [
-  { label: "From list", command: () => showAddByCodeList() },
-  { label: "From file", command: () => showAddByFile() }
-];
-
-const debounceTimer = ref(0);
 watch(ecl, newValue => {
   clearTimeout(debounceTimer.value);
   debounceTimer.value = window.setTimeout(async (): Promise<void> => {
     eclStatus.value = await EclService.validateECL(newValue);
-    if (eclStatus.value.valid) lastValidEcl.value = newValue;
+    if (eclStatus.value.valid) {
+      lastValidEcl.value = newValue;
+      eclAsQuery.value = await EclService.getQueryFromECL(newValue);
+    }
   }, 600);
 });
+watch(
+  () => cloneDeep(eclAsQuery.value),
```

Comment:
watch works deeply on reactive objects. The below should work just as well.
watch(eclAsQuery, () => ...)